content/sql/concepts/window-functions/terms/lag/lag.md

```diff
@@ -29,33 +32,106 @@ OVER ( [ partition_by ] order_by )
 - `partition_by` - Allows the result set to be grouped based on a column. This is an optional argument, if not specified the result set will be treated as a single group.
 - `order_by` - Determines the order of the result set. If `partition_by` is specified, it will order the grouped data instead.
 
-## Example
+## Examples
 
-The example below uses the `LAG()` function to create the new column `previous_age` based on values in the previous row of the `Users` table.
+The examples below use the table Sales, which contains orders of different products that are sold on a particular date and their price that they are sold to.
 
-Users Table
+Sales Table
 
-| first_name | last_name | age |
-| ---------- | --------- | --- |
-| chris      | smith     | 30  |
-| dave       | james     | 19  |
-| jenna      | black     | 35  |
-| kyle       | xy        | 60  |
+| product_id | order_date | sold_price |
+| ---------- | ---------- | -------- |
+|          1 | 2024-01-01 |      100 |
+|          1 | 2024-02-01 |      120 |
+|          1 | 2024-03-01 |       80 |
+|          2 | 2024-01-01 |       50 |
+|          2 | 2024-02-01 |       60 |
+|          2 | 2024-02-02 |       60 |
+
+Example 1 : 
+
+The below example uses `LAG()` on column `order_date` with Order By `order_date`.
 
 ```sql
-SELECT *,
-    LAG(age, 1) OVER (
-  ORDER BY age ASC) AS previous_age
-FROM Users;
+SELECT
+    product_id,
+    order_date,
+    sold_price,
+    LAG(order_date) OVER (
+      ORDER BY order_date
+      ) AS previous_order_date
+FROM
+    sales;
 ```
 
-The output is a table that features a new column `previous_age`, which holds the values from the previous records. The first record is null because a default was not specified and the previous row would be out of range.
+The output is a table with a new column `previous_order_date`, which contains previous entries of `order_date` in the table when ordered by `order_date`. The offset is set to 1 by default, and NULL is displayed when the offset is out of range.
```

Comment:
It's better to clarify the nature of the `LAG()` function's behavior when the offset is out of range, i.e., when the previous_order_date is NULL.

```suggestion
The output is a table with a new column `previous_order_date`, which contains previous entries of `order_date` in the table when ordered by `order_date`. The default offset is set to 1, meaning the previous row’s order_date will be displayed. When there is no previous row (such as for the first entry in the table), NULL will be displayed.
```